crates/rmf_site_editor/src/mapf_rse/negotiation/mod.rs

```diff
@@ -0,0 +1,360 @@
+                        entity_id_map: name_map
+                            .into_iter()
+                            .map(|(id, bits_string)| (id, bits_string_to_entity(&bits_string)))
+                            .collect(),
+                        error_message: None,
+                        conflicting_endpoints: Vec::new(),
+                    };
+                }
+            },
+        };
+    }
+}
+
+pub fn start_compute_negotiation(
+    mut commands: Commands,
+    locations: Query<(Entity, &NameInSite, &Point<Entity>), With<LocationTags>>,
+    anchors: Query<(Entity, &Anchor, &GlobalTransform)>,
+    negotiation_request: EventReader<NegotiationRequest>,
+    negotiation_params: Res<NegotiationParams>,
+    mut negotiation_debug_data: ResMut<NegotiationDebugData>,
+    mut compute_negotiation_task: Query<(Entity, &mut NegotiationTask)>,
+    current_level: Res<CurrentLevel>,
+    grids: Query<(Entity, &Grid)>,
+    child_of: Query<&ChildOf>,
+    robots: Query<(Entity, &NameInSite, &Pose, &Affiliation<Entity>), With<Robot>>,
+    robot_descriptions: Query<(&DifferentialDrive, &CircleCollision)>,
+    tasks: Query<(&RobotTask, &GoToPlace)>,
+) {
+    if negotiation_request.len() == 0 {
+        return;
+    }
+    if !compute_negotiation_task.is_empty() {
+        let Ok((entity, negotiation_task)) = compute_negotiation_task.single_mut() else {
+            error!("Error: Found multiple negotiation tasks!");
+            return;
+        };
+
+        if let NegotiationTaskStatus::InProgress { .. } = negotiation_task.status {
+            warn!("Negotiation requested while another negotiation is in progress");
+            return;
+        }
+        // Computation is done, we can despawn the previous negotiation task
+        commands.entity(entity).despawn();
+    }
+    negotiation_debug_data.selected_negotiation_node = None;
+
+    // Occupancy
+    let mut occupancy = HashMap::<i64, Vec<i64>>::new();
+    let mut cell_size = 1.0;
+    let grid = grids
+        .iter()
+        .filter_map(|(grid_entity, grid)| {
+            if let Some(level_entity) = current_level.0 {
+                if child_of
+                    .get(grid_entity)
+                    .is_ok_and(|co| co.parent() == level_entity)
+                {
+                    Some(grid)
+                } else {
+                    None
+                }
+            } else {
+                None
+            }
+        })
+        .next();
+    match grid {
+        Some(grid) => {
+            cell_size = grid.cell_size;
+            for cell in grid.occupied.iter() {
+                occupancy.entry(cell.y).or_default().push(cell.x);
+            }
+            for (_, column) in &mut occupancy {
+                column.sort_unstable();
+            }
+        }
+        None => {
+            occupancy.entry(0).or_default().push(0);
+            warn!("No occupancy grid found, defaulting to empty");
+        }
```

Comment:
I find this behavior a bit confusing.
I understand that if there is no occupancy grid it might not be very obvious how to generate one (you would need to send an event to another system and "temporarily pause" this one.
What I noticed however, is that if no occupancy grid is available and users ask to generate a plan, this warning will be printed, the plan will be generated according to an empty occupancy grid _and_ a new occupancy grid will be generated that makes the plan instantly invalid.
If using the newly generated occupancy grid for the plan is not possible I would recommend also not triggering the generation of a new one or the result can be quite confusing

Reply:
This is fixed by this PR https://github.com/open-rmf/rmf_site/pull/339, under point 2.